pkg/build/api/validation/validation_test.go

```diff
@@ -116,8 +150,19 @@ func TestBuildConfigValidationFailure(t *testing.T) {
 			},
 		},
 	}
-	if result := ValidateBuildConfig(buildConfig); len(result) != 2 {
-		t.Errorf("Unexpected validation result %v", result)
+	errors := ValidateBuildConfig(buildConfig)
+	if len(errors) != 1 {
+		t.Fatalf("Unexpected validation errors %v", errors)
+	}
+	err := errors[0].(*fielderrors.ValidationError)
+	if err.Type != fielderrors.ValidationErrorTypeInvalid {
+		t.Errorf("Unexpected error type, expected %s, got %s", fielderrors.ValidationErrorTypeInvalid, err.Type)
+	}
+	if err.Field != "triggers" {
+		t.Errorf("Unexpected field name expected name, got %s", err.Field)
+	}
+	if !strings.Contains(err.Detail, "only one ImageChangeTrigger is allowed") {
+		t.Errorf("Unexpected error details: %s", err.Detail)
```

Comment:
I don't like comparing specific error descriptions. This unnecessarily couples the test with the implementation.
However in this case checking that we've got a validation error on triggers is not enough, right?

Maybe using constants or error codes could help us. Constants for the error message seems better than error codes in this context. @soltysh what do you think?